web/src/main/java/org/akaza/openclinica/control/core/SecureController.java

```diff
@@ -1230,6 +1230,20 @@ public String getPropertiesDir() {
         absolutePath = absolutePath.replaceAll("placeholder.properties", "");
         return absolutePath;
     }
-    
+
+    protected static boolean isStudyLevelUser(HttpServletRequest request){
+        StudyBean currentStudy = (StudyBean) request.getSession().getAttribute("study");
+        UserAccountBean ub = (UserAccountBean) request.getSession().getAttribute(USER_BEAN_NAME);
+        ArrayList studyUserRoles = ub.getRoles();
+        boolean isStudyLevelUser = true;
+        for(int i = 0; i < studyUserRoles.size(); i++) {
+            StudyUserRoleBean studyUserRole = (StudyUserRoleBean) studyUserRoles.get(i);
+            if (studyUserRole.getStudyId() == currentStudy.getId() && currentStudy.getParentStudyId() > 0) {
+                isStudyLevelUser = false;
+                break;
+            }
+        }
+        return isStudyLevelUser;
```

Comment:
Rather than repeating the same code in two places, could you refactor it so it can be shared by both classes?

Reply:
Moved the method into a service class. and using that class in both `SecureController.java` and `CoreSecureController.java` having their own set of inheritors.